### ISSUE TITLE
Comment issues with {- and {+ on example file ProjectAdministration.adl

### ISSUE THREAD

**Issue by MvanRooijen** [AmpersandTarski/Ampersand#594]
I've downloaded the example files ProjectAdministration.adl and ProjectAdministration.xlsx
 when building the file in ampersand I get the following error:

`Lexer error at line 187:11, file ProjectAdministration.adl
  Unterminated PURPOSE section`

How can I fix this? I am still learning ampersand (Rule Based Design at Open University) and would like some files to get some practice in.

**Comment by RieksJ** [comment]
@MvanRooijen: changes in the most recent Ampersand version disallow the original syntax of the `PURPOSE` statement. `ProjectAdministration` had already partly been changed to conform to the new syntax, which currently is 
~~~
 PURPOSE <object> <name> {+<purposetext>+}
~~~
where it used to be
~~~
 PURPOSE <type> <id> {+ <purposetext> -}
~~~
You may need to upgrade to the most recent (development)version of Ampersand to get things right again.

@hanjoosten Perhaps it is a good idea to (temporarily) allow the old syntax again?

**Comment by dartheide** [comment]
That is confusing, version 373 (25-11-2016) for windows accept old syntax and 373 on debian build 18-12-2016 accept only the new syntax.

**Comment by hanjoosten** [comment]
Yes it is. Sorry for that. We made the decision to not have backwards compatibility here, because the -} would break block comments, which is very annoying. This in combination with the still relatively small user group we have, we thought this was acceptable. Another advantage is that we have all such problems in a small period of time.

**Comment by MvanRooijen** [comment]
Ampersand-v3.7.3 [HEAD:d9e3dd3], build time: 25-Nov-16 16:02:42 Coordinated Universal Time gives the same error.

Please advise on which version to install so I can have a working example.
Either that or a more in detail explanation on how to fix the example to make it work in the current version

**Comment by MvanRooijen** [comment]
Rien, thank you for the updated example, will give it a go.
Are there many such differences between the "old" and "new" syntax, and can I find some more details on those differences?

Note I am nowhere near a programmer, but we are required to deliver a working non error generating working Ampersand script to pass the course

**Comment by RieksJ** [comment]
The differences are slight, and easy to accommodate. 
Basically, the new syntax has nicely paired brackets, which the old syntax did not have.

So, if you have a `PURPOSE` (or something else that was already followed by a `{+` token), then this `{+` token used to have a corresponding `-}`. This corresponding `-}` should now be replaced with a `+}`, so every `{+` is then matched with a `+}`. 

Of course, any other comment block (e.g. comment that starts with a `{-` token) should be terminated with a `-}` token.

If you consider this issue complete, please close the issue.

**Comment by MvanRooijen** [comment]
Unfortunately the issue is not resolved. Might have something to do with the version I am running (I upgraded it to the Ampersand-v3.7.3 [HEAD:d9e3dd3], build time: 25-Nov-16 16:02:42 Coordinated Universal Time however.

In which version did you test the improved ProjectAdministration.adl?
I'll download that instead.

I am very curious to see what the software does when working (the files I checked in the https://github.com/AmpersandTarski/ampersand-models URL are not generating anything without errors however).

**Comment by hanjoosten** [comment]
Next friday, the release of ampersand v3.8.0 is sceduled. The new `{+` ... `+}` syntax will be in it. If you cannot wait, you could build the developement branch for yourself. Installation instructions are in the documentation.

**Comment by RieksJ** [comment]
If you can't wait, here's a (temporarily working) link: [ampersand.exe](https://dl.dropboxusercontent.com/u/39690625/ampersand.exe) (version Ampersand-v3.7.3 [development-c2fabc7])

**Comment by dartheide** [comment]
A bit of topic does the new windows version also accept the --add-semantic-metamodel switch?

**Comment by RieksJ** [comment]
The windows version is compiled from the same sources as the other versions (e.g. Linux). Only working with the windows version myself, I do expect all these versions to have the same functionality, including switches. Bugs, e.g. w.r.t. switches would therefore appear in Ampersand regardless of the OS it is compiled for.

**Comment by hanjoosten** [comment]
I have released version 3.8.0 today. This works with the projectadministration in the master branch of ampersand-models.

**Comment by hanjoosten** [comment]
I've closed this issue because it should be solved with the new release. If you still have trouble, don't hesitate to reopen it.
